cirq-core/cirq/sim/simulator_base.py

```diff
@@ -229,9 +214,7 @@ def _core_iterator(
                 except TypeError:
                     raise TypeError(f"{self.__class__.__name__} doesn't support {op!r}")
 
-            step_result = self._create_step_result(sim_state)
-            yield step_result
-            sim_state = step_result._sim_state
+            yield self._create_step_result(sim_state)
```

Comment:
@95-martin-orion note this also finally reverses the change that concerned you here: https://github.com/quantumlib/Cirq/pull/4110#pullrequestreview-702346996